src/env.rs

```diff
@@ -308,12 +336,18 @@ impl Source for Environment {
             };
 
             m.insert(key, Value::new(Some(&uri), value));
+
+            Ok(())
         };
 
         match &self.source {
-            Some(source) => source.clone().into_iter().for_each(collector),
-            None => env::vars().for_each(collector),
-        }
+            Some(source) => source
+                .clone()
+                .into_iter()
+                .map(|(key, value)| (OsString::from(key), OsString::from(value)))
```

Comment:
I don't have a good idea on how to avoid it. On Linux, at least, this should be zero cost as the underlying `Vec<u8>` is just moved... but on Windows it might not be

Reply:
On Windows, an `OsString` is still a `Vec<u8>` and is full.  The documentation for the `From` impl even says

> This conversion does not allocate or copy memory.

Reply:
Cool. I thought it'd be u16 to match Windows UTF-16, but you're right :) Resolving this, then.